arlas-tagger-core/src/main/java/io/arlas/tagger/service/TagExploreService.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to Gisaïa under one or more contributor
+ * license agreements. See the NOTICE.txt file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Gisaïa licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.arlas.tagger.service;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.arlas.tagger.app.ArlasTaggerConfiguration;
+import io.arlas.tagger.kafka.TagKafkaConsumer;
+import io.arlas.tagger.model.request.TagRefRequest;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public class TagExploreService {
+    private Logger LOGGER = LoggerFactory.getLogger(TagExploreService.class);
+    private final ArlasTaggerConfiguration configuration;
+    private KafkaConsumer consumer;
+    protected static ObjectMapper MAPPER = new ObjectMapper();
+
+    public TagExploreService(ArlasTaggerConfiguration configuration, String topic, String consumerGroupId, Integer batchSize) {
+        this.configuration = configuration;
+        this.consumer = TagKafkaConsumer.build(configuration, topic, consumerGroupId, batchSize, true);
+    }
+
+    public List<TagRefRequest> getTagRefList() {
+        List<TagRefRequest> results = new ArrayList<>();
+
+        consumer.seekToBeginning(consumer.assignment());
+        ConsumerRecords<String, String> records = consumer.poll(Duration.ofMillis(configuration.kafkaConfiguration.consumerPollTimeout));
+        while (!records.isEmpty()) {
```

Comment:
You could maybe avoid the double assignement of `records` with `while(!(records = consumer.poll(Duration.ofMillis(configuration.kafkaConfiguration.consumerPollTimeout))).isEmpty())`

Reply:
Done.